migration/workspace-query-separation/src/main/java/ddf/migration/workspace/query/separation/WorkspaceQueryMigration.java

```diff
@@ -0,0 +1,226 @@
+      QueryRequest queryRequest = new QueryRequestImpl(query);
+
+      QueryResponse queryResponse = catalogFramework.query(queryRequest);
+
+      queryResponse
+          .getResults()
+          .stream()
+          .map(Result::getMetacard)
+          .forEach(this::migrateWorkspaceMetacard);
+
+    } catch (UnsupportedQueryException | SourceUnavailableException | FederationException e) {
+      LOGGER.debug("An unexpected error occurred during migration", e);
+    }
+
+    LOGGER.info("Completed workspace query data migration");
+    System.out.println("Completed: Workspace Query Data Migration");
+  }
+
+  private void migrateWorkspaceMetacard(Metacard workspaceMetacard) {
+    LOGGER.info("Migrating workspace metacard with id [{}]", workspaceMetacard.getId());
+
+    final Attribute queriesAttribute = workspaceMetacard.getAttribute(WORKSPACE_QUERIES);
+
+    if (queriesAttribute == null || queriesAttribute.getValues().isEmpty()) {
+      LOGGER.info(
+          "Workspace metacard with id [{}] did not contain query data", workspaceMetacard.getId());
+      return;
+    }
+
+    final List<Serializable> queriesAsXml = queriesAttribute.getValues();
+
+    LOGGER.info(
+        "Beginning migration of {} query metacards from workspace [{}]",
+        queriesAsXml.size(),
+        workspaceMetacard.getId());
+
+    List<Metacard> queryMetacards =
+        queriesAsXml
+            .stream()
+            .filter(String.class::isInstance)
+            .map(String.class::cast)
+            .map(xml -> xml.getBytes(StandardCharsets.UTF_8))
+            .map(ByteArrayInputStream::new)
+            .map(this::xmlToMetacard)
+            .filter(Objects::nonNull)
+            .collect(Collectors.toList());
+
+    LOGGER.info("Creating query metacards for workspace [{}]", workspaceMetacard.getId());
+
+    CreateResponse response = createMetacards(queryMetacards);
+
+    if (response != null) {
+      List<String> createdMetacardIds =
+          response.getCreatedMetacards().stream().map(Metacard::getId).collect(Collectors.toList());
+
+      LOGGER.info("Created query metacards with IDs {}", createdMetacardIds);
+
+      workspaceMetacard.setAttribute(
+          new AttributeImpl(WORKSPACE_QUERIES, (Serializable) createdMetacardIds));
+      updateMetacard(workspaceMetacard);
+
+      LOGGER.info(
+          "Updated workspace metacard [{}] with query metacard associations {}",
+          workspaceMetacard.getId(),
+          createdMetacardIds);
+    }
+  }
+
+  private UpdateResponse updateMetacard(Metacard metacard) {
+    return AccessController.doPrivileged(
+        (PrivilegedAction<UpdateResponse>)
+            () -> {
+              try {
+                return catalogFramework.update(new UpdateRequestImpl(metacard.getId(), metacard));
+              } catch (IngestException | SourceUnavailableException e) {
+                LOGGER.debug("Error updating workspace metacard [{}]", metacard.getId());
```

Comment:
Consider making this error message more specific to the exception being thrown

Reply:
Added the exception to the log message so that it's more specific